core/benches/scheduler.rs

```diff
@@ -0,0 +1,494 @@
+        ixs.push(ComputeBudgetInstruction::set_loaded_accounts_data_size_limit(TAG_NUMBER));
+        let message = Message::new(&ixs, Some(&payer.pubkey()));
+        let tx = Transaction::new(&[payer], message, Hash::default());
+        RuntimeTransaction::from_transaction_for_tests(tx)
+    }
+
+    fn build(
+        &self,
+        count: usize,
+        is_single_payer: bool,
+    ) -> Vec<RuntimeTransaction<SanitizedTransaction>> {
+        let mut transactions = Vec::with_capacity(count);
+        // non-contend low-prio tx is first received
+        transactions.push(self.build_tracer_transaction());
+
+        let compute_unit_price = 1_000;
+        let single_payer = Keypair::new();
+        for _n in 1..count {
+            let payer = if is_single_payer {
+                // recreate keypair from single_payer
+                Keypair::from_bytes(&single_payer.to_bytes()).expect("Failed to create Keypair")
+            } else {
+                Keypair::new()
+            };
+            let mut ixs = self.prepare_instructions(&payer);
+            let prioritization =
+                ComputeBudgetInstruction::set_compute_unit_price(compute_unit_price);
+            ixs.push(prioritization);
+            let message = Message::new(&ixs, Some(&payer.pubkey()));
+            let tx = Transaction::new(&[payer], message, Hash::default());
+            let transaction = RuntimeTransaction::from_transaction_for_tests(tx);
+
+            transactions.push(transaction);
+        }
+
+        transactions
+    }
+
+    fn prepare_instructions(&self, payer: &Keypair) -> Vec<Instruction>;
+}
+
+struct SimpleTransferTxBuilder;
+impl TestTxsBuilder for SimpleTransferTxBuilder {
+    fn prepare_instructions(&self, payer: &Keypair) -> Vec<Instruction> {
+        let to_pubkey = Pubkey::new_unique();
+        vec![system_instruction::transfer(&payer.pubkey(), &to_pubkey, 1)]
+    }
+}
+
+struct ManyTransfersTxBuilder;
+impl TestTxsBuilder for ManyTransfersTxBuilder {
+    fn prepare_instructions(&self, payer: &Keypair) -> Vec<Instruction> {
+        const MAX_TRANSFERS_PER_TX: usize = 35;
+        let to = Vec::from_iter(
+            std::iter::repeat_with(|| (Pubkey::new_unique(), 1)).take(MAX_TRANSFERS_PER_TX),
+        );
+        system_instruction::transfer_many(&payer.pubkey(), &to)
+    }
+}
+
+struct PackedNoopsTxBuilder;
+impl TestTxsBuilder for PackedNoopsTxBuilder {
+    fn prepare_instructions(&self, _payer: &Keypair) -> Vec<Instruction> {
+        // Creating noop instructions to maximize the number of instructions per
+        // transaction. We can fit up to 355 noops.
+        const MAX_INSTRUCTIONS_PER_TRANSACTION: usize = 355;
+        let program_id = Pubkey::new_unique();
+        (0..MAX_INSTRUCTIONS_PER_TRANSACTION)
+            .map(|_| Instruction::new_with_bytes(program_id, &[], vec![]))
+            .collect()
+    }
+}
+
+struct BenchContainer<Tx: TransactionWithMeta> {
```

Comment:
Not sure I understand the purpose of the `BenchContainer`, we can just have a free function for `fill_container` right?

Reply:
left over stuff, removed it @ b85b87e99fb7a5b780123707a767e8bda3d6e977